### DISCUSSION TITLE
Recommend trying Will Pirkle's diode ladder filter emulation and a diode clipper.

### DISCUSSION THREAD

**Discussion by 01GOD** [copych/AcidBox#20] [Ideas]
Hi, 

Before Apple removed the nice native Apple platform apps and AU plugins (that I had worked A LOT to make so nice) from the Apple app stores, an app I made called ELECTROBASE was thankfully getting popular. Sadly Apple removed that app after the charitable foundation I manage promoted liberation of Palestine. And sadly Apple also removed all of the other apps and AU plugins I had worked A LOT to make so nice, and also suspended the app distribution membership of the charitable foundation I manage. That happened more than a calendar year ago (I'm thankfully a trend setter:), and, regardless of Apple saying that would only be 1 calendar year suspension, the charitable foundation I manage still hasn't been allowed back on the Apple app stores. Obviously Apple (cr-Apple) has had an illegal monopoly (made possible by a buffer of corrupt attorneys) on iOS and iPadOS app stores, and sadly the projected app sales Apple stole from the charitable foundation I manage the past calendar year is at this point in the thousands of US DOLLARS range. So everybody reading this please BOYCOTT Apple!!!

Anyway, had to convey that firstly, as ELECTROBASE is thankfully an AWESOME app and israel (and "israel affiliates") have stolen hours of happiness from MANY people around the World by preventing them from having access to those thankfully fun and enjoyable music apps I made. That said, My favorite diode ladder filter emulation is the emulation thankfully shared by Will Pirkle. 

So I recommend that filter for proper filter response (it DEFINITELY required some tuning though, as the reso on that filter tends to go into whistling mode fast past a certain level of resonance) and also recommend adding a diode clipper (if there isn't one in there already). That combo provides the standard MXR style sound that many seem to like most. 

Thanks for posting this nice project!!! I may climb out of the bed of somber sobriety about the rampant corruption in much of the human species (and the heavily inbred genocidal subhuman bipeds I mentioned earlier) to try that on an ESP32 soon.

**Comment by copych** [comment]
Thanks for your comment. As you might have noticed, this is a hobby for me, I am not either a pro-coder or an IT-businessman, or even an indie-publisher. I just wanna share results of my efforts with persons who make some fun with it or even create something worth, or maybe contribute something into the projects.
Thank you for the recommended filter and limitter. I will check it soon.
On the policy topics mentioned. The world is just half-step close to a self-termination, and if someone asked me, I would vote for Palestine to be recognized as a sovereign state. Also I would vote for dissolving one well known Treaty Organization, as IT IS the firestarter. But it's a GitHub.
PS Apple was never a choice for me, frankly. I don't know why. Maybe because every creative person, they say, should have an iPhone )))

**Comment by 01GOD** [comment]
Glad to hear agree about Palestine! Pretty obvious that is the right
choice.

Yes, it is github, and thus a great place to notify other software
engineers when there is corruption in the companies that run the app stores
people work to put both free and commercial apps on.

Mainly replying to note that a “diode clipper” is generally considered a
type of distortion, although yes it does function like a limiter in a way.
Mainly clips the top and bottom  of the wave to make those flat.

**Reply by copych** [reply]
Yep, I understand the difference between a clipper and a limitter, my mistake, mainly cause I'm not a native English speaker.
I have looked through the code of Diode Ladder Filter by Will Pirkle, and now doubt if ESP32 @240MHz have enough CPU cycles between i2s writes. But I'll check that.

**Comment by 01GOD** [comment]
Ah, hopefully it does! Was admittedly surprised had made that other moog
style filter emulation sound as similar as it does. Nice job on that.

**Comment by copych** [comment]
I'd tried 3 or 4 more filters before I've found that open303 combination. Guys did a lot of research.

**Comment by 01GOD** [comment]
I haven’t heard the range of the CO and resonance and accent knobs yet, so
don’t know how those sound. I know that Will Pirkle filter emulation is
supposedly bases on the diode ladder filter’s transfer function (frequency
gain plot mainly) and it sounds proper generally.

A diode clipper seems to be a pretty simple algorithm in comparison, so
that is likely easiee on processor.

**Reply by copych** [reply]
https://www.kvraudio.com/forum/viewtopic.php?t=262829 guys analyzed the schematics and added a feedback chain into the filter
